soc/arm/nxp_imx/rt5xx/power.c

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/pm/pm.h>
+#include <zephyr/drivers/pinctrl.h>
+#include "fsl_power.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+/*!< Power down all unnecessary blocks */
+#define NODE_ID DT_INST(0, nxp_pdcfg_power)
+#define EXCLUDE_FROM_DEEPSLEEP ((const uint32_t[]) \
+					DT_PROP_OR(NODE_ID, deep_sleep_config, {}))
+
+#define EXCLUDE_FROM_DEEP_POWERDOWN ((const uint32_t[]){0, 0, 0, 0})
+
+/* System clock frequency. */
+extern uint32_t SystemCoreClock;
+
+#define FLEXSPI_DEVICE DT_NODELABEL(flexspi)
```

Comment:
I think for these lines you may want to also check that pin control nodes are present for the FlexSPI device. Otherwise I believe that `PINCTRL_DT_DEFINE()` may result in a build error.  `DT_NODE_HAS_PROP(FLEXSPI_DEVICE, pinctrl)` should work for this

Reply:
i have tried by deleting the `pinctrl` properties deleted and do not see any build failures.